### PR TITLE
ref(btn icon): adding alt icon prop

### PR DESCRIPTION
**Pull request by doralchan**
**Before:**
Screen Shot 2020-02-25 at 2 48 25 PM

**After:**
Screen Shot 2020-02-25 at 2 47 29 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/button.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ type Props = {
   busy?: boolean;
   to?: string | object;
   href?: string;
-  icon?: string;
+  icon?: string | object;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/button.tsx]: Is the object form a react node? If so you should use React.ReactNode.

File: src/sentry/static/sentry/app/components/button.tsx [hunk 2]
```diff
@@ -163,7 +163,7 @@ class Button extends React.Component<ButtonProps, {}> {
           priority={priority}
           borderless={borderless}
         >
-          {icon && (
+          {icon && typeof icon === 'string' ? (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/button.tsx]: Ah, looking at the snapshots, it looks like the "else" condition will pick up the case where icon is falsey, we could move the `typeof icon` check to inside of `<Icon>`

File: src/sentry/static/sentry/app/components/button.tsx (hunk 1 above)
Comment (billyvg) [inline src/sentry/static/sentry/app/components/button.tsx]: Does ts get mad if we be more strict with the type and change `object` to `React.Element`?

Comment (doralchan) [reply src/sentry/static/sentry/app/components/button.tsx]: Yes it does get mad

Comment (doralchan) [reply src/sentry/static/sentry/app/components/button.tsx]: This works

**Review by markstory — APPROVED** [review]
I like this approach.  It will let us incrementally update buttons and not have to style every icon individually.

**Review by billyvg — APPROVED** [review]

**Pull request merged**